src/customJs/tools/promo_code/types.ts

```diff
@@ -1,10 +1,28 @@
 import { EMPTY_SELECTION } from '../../constants';
 import { Alignment, PropertyGroupsFrom, ToolValuesFrom } from '../../types';
 
+export type PromoCodeTypes = 'percent' | 'money' | 'shipping';
+
 export type PromoCodeBase = {
-  promo_code: {
+  promo_code_store: {
     store: string | EMPTY_SELECTION;
-    promo_code: string | EMPTY_SELECTION;
+    isDynamic: boolean;
+  };
+  promo_code: {
+    code: string | EMPTY_SELECTION;
+  };
+  dynamic_code: {
+    code_type: PromoCodeTypes;
+    code_value: string;
```

Comment:
These values could simply be named `type` and `value`, right? They represent the `value` and `type` of a dynamic promo code.

Reply:
update code_type by type and code_value by amount